### PR TITLE
Fix RobertaFeaturizer #3476

### PR DESCRIPTION
**Pull request by quincylin1**
## Description

Fix https://github.com/deepchem/deepchem/issues/3476

Convert `datapoints` to `List` if it is a `pd.Series` and add `padding` to pad the embeddings to the same length 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by arunppsg** [comment]
Hey, it seems there are some failing tests (see [here](https://github.com/deepchem/deepchem/actions/runs/8299238434/job/22714480583?pr=3898#step:14:804)). Could you take a look at it?

**Comment by quincylin1** [comment]
> Hey, it seems there are some failing tests (see [here](https://github.com/deepchem/deepchem/actions/runs/8299238434/job/22714480583?pr=3898#step:14:804)). Could you take a look at it?

Modified the unit tests in `test_roberta_tokenizer.py` slightly. Please let me know if there's any issue :)

**Comment by arunppsg** [comment]
The changes looks good to me and the failures are not related to this PR. I request @rbharath for a second pass.

**Review by arunppsg — APPROVED** [review]

File: deepchem/feat/roberta_tokenizer.py [hunk 1]
```diff
@@ -57,8 +58,14 @@ def _featurize(self, datapoint: str, **kwargs) -> List[List[int]]:
...
 
         # the encoding is natively a dictionary with keys 'input_ids' and 'attention_mask'
-        encoding = list(self(datapoint, **kwargs).values())
+        encoding = list(super().__call__(datapoint, **kwargs).values())
         return encoding
 
-    def __call__(self, *args, **kwargs) -> Dict[str, List[int]]:
-        return super().__call__(*args, **kwargs)
+    def __call__(self,
+                 datapoints,
```
Comment (rbharath) [inline deepchem/feat/roberta_tokenizer.py:65]: Can you add a type annotation here?

File: deepchem/feat/roberta_tokenizer.py [hunk 2]
```diff
@@ -57,8 +58,14 @@ def _featurize(self, datapoint: str, **kwargs) -> List[List[int]]:
...
+                 padding: bool = True,
+                 **kwargs) -> Dict[str, List[int]]:
```
Comment (rbharath) [inline deepchem/feat/roberta_tokenizer.py:67]: Given we are not directly calling the superclass, let's add a brief docstring here

File: deepchem/feat/tests/test_roberta_tokenizer.py [hunk 3]
```diff
@@ -16,8 +16,7 @@ def test_smiles_call():
                                 add_special_tokens=True,
                                 truncation=True)
     for emb in [embedding, embedding_long]:
-        assert 'input_ids' in emb.keys() and 'attention_mask' in emb.keys()
```
Comment (rbharath) [inline deepchem/feat/tests/test_roberta_tokenizer.py:19]: Why are we removing asserts here?

**Review by rbharath — COMMENTED** [review]
I have a few questions on the changes below

File: deepchem/feat/roberta_tokenizer.py (hunk 1 above)
Comment (quincylin1) [reply deepchem/feat/roberta_tokenizer.py]: Please see the latest commit

File: deepchem/feat/roberta_tokenizer.py (hunk 2 above)
Comment (quincylin1) [reply deepchem/feat/roberta_tokenizer.py]: Please see the latest commit

File: deepchem/feat/tests/test_roberta_tokenizer.py (hunk 3 above)
Comment (quincylin1) [reply deepchem/feat/tests/test_roberta_tokenizer.py:19]: Because the `__call__` function in `RobertaFeaturizer` now returns the `input_ids` only, which are the actual embeddings of the molecules. The reason is that `_featurize_shard()` in`CSVLoader` is later called, in which the `features` List needs the embeddings only but not the attention masks (see [here](https://github.com/deepchem/deepchem/blob/eef32b047ac69a32093d8a51bca3a1283e2b2112/deepchem/data/data_loader.py#L430)). This is actually why I modified the `__call__` in `RobertaFeaturizer` so that it returns the input_ids only, unlike the `__call__` in the inherited HuggingFace's `RobertaTokenizerFast`, which returns the whole dict (input_ids and attention_mask). 

So the asserts are removed since `attention_mask` are not returned anymore, and the test asserts that the lists of `input_ids` for all the molecules all have the same length after specifying `padding=True`.

**Comment by quincylin1** [comment]
@arunppsg @rbharath any chance to review this PR? Thanks!
